oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java

```diff
@@ -334,13 +338,20 @@ public IdToken idTokenWithAudience(String targetAudience, List<IdTokenProvider.O
     return IdToken.create(rawToken);
   }
 
-  private HttpResponse getMetadataResponse(String url) throws IOException {
+  private HttpResponse getMetadataResponse(String url, RequestType requestType) throws IOException {
```

Comment:
I know I told you to swap from UNSPECIFIED to UNTRACKED, but from reading the comments below as additional context, I think my opinion has slightly changed and I'm flip-flopping (sorry!). Let me know what you think. 

To me, it seems a bit odd that CredentialsMetrics.DO_NOT_SEND does not send the CRED_TYPE and RequestType.UNSPECIFIED does not send the AUTH_REQUEST_TYPE, but RequestType.UNTRACKED doesn't send both (it doesn't send the header at all). I would think that CredentialMetrics only impacts CRED_TYPE and RequestType for AUTH_REQUEST_TYPE.

What if we change the calls to something like this:
`getMetadataResponse(buildHttpRequest(url, requestType, shouldAddMetricsHeader));`

We add a new helper function `buildHttpRequest(...)` that takes in an additional bool flag where `universe_domain` and `getAccount` calls set the flag to be false. The flag gates adding the MetricsHeader.

Reply:
I don't feel too strongly about this, will make the change.